plugin/evm/gossiper_orders.go

```diff
@@ -0,0 +1,166 @@
+package evm
+
+import (
+	"time"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/subnet-evm/plugin/evm/message"
+	"github.com/ava-labs/subnet-evm/plugin/evm/orderbook"
+	"github.com/ava-labs/subnet-evm/plugin/evm/orderbook/hubbleutils"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+func (n *pushGossiper) GossipSignedOrders(orders []*hubbleutils.SignedOrder) error {
+	select {
+	case n.ordersToGossipChan <- orders:
+	case <-n.shutdownChan:
+	}
+	return nil
+}
+
+func (n *pushGossiper) awaitSignedOrderGossip() {
+	n.shutdownWg.Add(1)
+	go executeFuncAndRecoverPanic(func() {
+		var (
+			gossipTicker = time.NewTicker(ordersGossipInterval)
+		)
+		defer func() {
+			gossipTicker.Stop()
+			n.shutdownWg.Done()
+		}()
+
+		for {
+			select {
+			case <-gossipTicker.C:
+				if attempted, err := n.gossipSignedOrders(); err != nil {
+					log.Warn(
+						"failed to send signed orders",
+						"len(orders)", attempted,
+						"err", err,
+					)
+				}
+			case orders := <-n.ordersToGossipChan:
+				for _, order := range orders {
+					orderHash, err := order.Hash()
+					if err != nil {
+						log.Error("failed to hash order", "err", err)
+						continue
+					}
+					n.ordersToGossip[orderHash] = order
+				}
+				if attempted, err := n.gossipSignedOrders(); err != nil {
+					log.Warn(
+						"failed to send signed orders",
+						"len(orders)", attempted,
+						"err", err,
+					)
+				}
+			case <-n.shutdownChan:
+				return
+			}
+		}
+	}, "panic in awaitSignedOrderGossip", orderbook.AwaitSignedOrdersGossipPanicsCounter)
+
+}
+
+func (n *pushGossiper) gossipSignedOrders() (int, error) {
+	//
+	if (time.Since(n.lastOrdersGossiped) < minGossipOrdersBatchInterval) || len(n.ordersToGossip) == 0 {
+		return 0, nil
+	}
+	n.lastOrdersGossiped = time.Now()
+	orders := []*hubbleutils.SignedOrder{}
+	for orderHash, order := range n.ordersToGossip {
+		orders = append(orders, order)
+		delete(n.ordersToGossip, orderHash)
```

Comment:
does this have the affect that each node will gossip any 1 order only once?

Reply:
yes, right now the rpc node sends it out to all other nodes exactly once. Also, when a node receives a signed order message, it just saves it in the db and it doesn't forward it further

i checked this using metrics while testing on local

Reply:
For now it sounds good but we might need a more sophisticated gossip strategy along with retries maybe

Reply:
how about this - every node when it receives the message will also gossip after saving the order. There will be duplicates like this, but it's quite simple, and will solve for 2 cases:

1. when 1 node is not connected directly to all the other nodes, i.e. not all nodes are peers
2. when a node couldn't receive a message due to network issues/some other unforeseen reasons

Reply:
Yes this sounds good.

Reply:
done this now